ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/Rejections.scala

```diff
@@ -70,21 +73,77 @@ private[transaction] class Rejections(metrics: Metrics) {
     builder
       .setSubmitterInfo(transactionEntry.submitterInfo)
 
-    rejection.toStateV1RejectionReason match {
-      case RejectionReasonV0.Inconsistent(reason) =>
-        builder.setInconsistent(Inconsistent.newBuilder.setDetails(reason))
-      case RejectionReasonV0.Disputed(reason) =>
-        builder.setDisputed(Disputed.newBuilder.setDetails(reason))
-      case RejectionReasonV0.ResourcesExhausted(reason) =>
-        builder.setResourcesExhausted(ResourcesExhausted.newBuilder.setDetails(reason))
-      case RejectionReasonV0.PartyNotKnownOnLedger(reason) =>
-        builder.setPartyNotKnownOnLedger(PartyNotKnownOnLedger.newBuilder.setDetails(reason))
-      case RejectionReasonV0.SubmitterCannotActViaParticipant(details) =>
+    rejection match {
```

Comment:
Can we add some tests that check this in combination with _Conversions.scala_? I am thinking some table-driven checks that verify that `Rejection.Xyz` converts to `Status.of(code1, message1)`, etc.

Reply:
Yes, this is the next step.
Actually plan on moving this code to conversions as well so that it's in the same place, and then add tests for 2-way conversions. 

On another note I was also thinking of using these conversions as part of the `RejectionReasonUpdateNormalizer`, this would at least add some form of validation on the rejection as well.